plugins/importexport/native/NativeImportExportPlugin.inc.php

```diff
@@ -13,408 +13,71 @@
-								foreach ($foundWarningMessages as $foundWarningMessage) {
-									echo '- ' . $foundWarningMessage . "\n";
-								}
-							}
-						}
-					}
-				}
-
-				// Are there any import errors? Display them.
-				$foundErrors = false;
-				foreach ($errorTypes as $assocType => $localeKey) {
-					$currentErrors = $deployment->getProcessedObjectsErrors($assocType);
-					if (!empty($currentErrors)) {
-						echo __('plugins.importexport.common.errorsOccured') . "\n";
-						$i = 0;
-						foreach ($currentErrors as $currentErrorMessages) {
-							if (count($currentErrorMessages) > 0) {
-								echo ++$i . '.' . __($localeKey) . "\n";
-								foreach ($currentErrorMessages as $currentErrorMessage) {
-									echo '- ' . $currentErrorMessage . "\n";
-								}
-							}
-						}
-						$foundErrors = true;
-					}
-				}
-				// If there are any data or validataion errors
-				// delete imported objects.
-				if ($foundErrors || !empty($validationErrors)) {
-					// remove all imported issues and sumissions
-					foreach (array_keys($errorTypes) as $assocType) {
-						$deployment->removeImportedObjects($assocType);
-					}
-				}
-				return;
-			case 'export':
-				$outputDir = dirname($xmlFile);
-				if (!is_writable($outputDir) || (file_exists($xmlFile) && !is_writable($xmlFile))) {
-					echo __('plugins.importexport.common.cliError') . "\n";
-					echo __('plugins.importexport.common.export.error.outputFileNotWritable', array('param' => $xmlFile)) . "\n\n";
-					$this->usage($scriptName);
-					return;
-				}
-				if ($xmlFile != '') switch (array_shift($args)) {
-					case 'monograph':
-					case 'monographs':
-						file_put_contents($xmlFile, $this->exportSubmissions(
-							$args,
-							$press,
-							null,
-							$opts
-						));
-						return;
-				}
-				break;
-		}
-		$this->usage($scriptName);
+		return $filter;
 	}
 
 	/**
-	 * @copydoc ImportExportPlugin::usage
+	 * @see ImportExportPlugin::getAppSpecificDeployment
 	 */
-	function usage($scriptName) {
-		fatalError('Not implemented.');
+	function getAppSpecificDeployment($journal, $user) {
+		return new NativeImportExportDeployment($journal, $user);
 	}
 
 	/**
-	 * Pull out getopt style long options.
-	 * @param &$args array
-	 * #param $optCodes array
+	 * @see PKPNativeImportExportPlugin::loadCLILocales
 	 */
-	function parseOpts(&$args, $optCodes) {
-		$newArgs = [];
-		$opts = [];
-		$sticky = null;
-		foreach ($args as $arg) {
-			if ($sticky) {
-				$opts[$sticky] = $arg;
-				$sticky = null;
-				continue;
-			}
-			if (!starts_with($arg, '--')) {
-				$newArgs[] = $arg;
-				continue;
-			}
-			$opt = substr($arg, 2);
-			if (in_array($opt, $optCodes)) {
-				$opts[$opt] = true;
-				continue;
-			}
-			if (in_array($opt . ":", $optCodes)) {
-				$sticky = $opt;
-				continue;
-			}
-		}
-		$args = $newArgs;
-		return $opts;
+	function loadCLILocales() {
+		parent::loadCLILocales();
+		AppLocale::requireComponents(LOCALE_COMPONENT_APP_MANAGER);
```

Comment:
If the locale keys get moved to pkp-lib, this might be removable.

Reply:
Removed (from all contexts along with the `loadCLILocales` function